Remove pylint

potiuk opened the pull request:
We've agreed during the voting process that Pylint support
should be disabled: https://lists.apache.org/thread.html/r9e2cc385db8737ec0874ad09872081bd083593ee29e8303e58d21efb%40%3Cdev.airflow.apache.org%3E

This PR:

* removes all # pylint comments
* removes pylint pre-commits and related scripts/files
* removes CI jobs running pylint checks
* removes documentation about pylint
* removes unnecessary #noga (adds pre-commit for that)
* fixes some remaining `pydocstyle` errors after removing #noqa's

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

mik-laj commented on airflow/api/client/json_client.py:
```diff
@@ -31,12 +31,12 @@ def _request(self, url, method='GET', json=None):
             try:
                 data = resp.json()
-            except Exception:  # noqa pylint: disable=broad-except
+            except Exception:  # noqa
```
I suspect that this `noqa` tag has no effect on any tool now.  We should use [`yesqa`](https://github.com/asottile/yesqa) to delete the unused tag, but we can do it as a separate PR.

potiuk replied:
I think flake also complains about this case. But yeah. yesqa is a good idea. I will run it now no problem.

mik-laj replied:
Flake dont support this case. In other place, we have broad exception without `noqa` tag.

ashb commented on .pre-commit-config.yaml:
```diff
@@ -644,12 +642,6 @@ repos:
         files: ^airflow/www/static/js/
         entry: scripts/ci/static_checks/www_lint.sh
         pass_filenames: false
-      - id: bats-in-container-tests
-        name: Run in container bats tests
-        language: system
-        entry: ./scripts/ci/pre_commit/pre_commit_in_container_bats_test.sh
-        files: ^tests/bats/in_container/.*\.bats$|^scripts/in_container/.*sh
-        pass_filenames: false
```
Seems unrelated to pylint?

ashb replied:
Oh it moved.

potiuk replied:
It's moved to after the `build` step - the check failed if the build has not been run before (discovered it while testing this change). I can separate it out, but since we are changing the pre-commit anyway quite heavily, fixing it in the "big" pylint cleanup PR might be a good idea.

ashb replied:
Nah, it's good.

ashb commented on airflow/_vendor/connexion/decorators/parameter.py:
```diff
@@ -6,7 +6,7 @@
 
 from ..http_facts import FORM_CONTENT_TYPES
-from ..lifecycle import ConnexionRequest  # NOQA
+from ..lifecycle import ConnexionRequest
```
Generally I'd say we shouldn't check/change code under `airflow/_vendor/` -- so we should exclude this from the Yesqa check if it's complaining and not otherwise touch these files

potiuk replied:
Good point. fixing it

potiuk committed:
Removes pylint from our toolchain

We've agreed during the voting process that Pylint support
should be disabled: https://lists.apache.org/thread.html/r9e2cc385db8737ec0874ad09872081bd083593ee29e8303e58d21efb%40%3Cdev.airflow.apache.org%3E

This PR:

* removes all # pylint comments
* removes pylint pre-commits and related scripts/files
* removes CI jobs running pylint checks
* removes documentation about pylint
* removes unnecessary #noga (adds pre-commit for that)
* fixes some remaining pydocstyle errors after removing #noqa's

kaxil reviewed (approved):
![](https://media.giphy.com/media/BpGWitbFZflfSUYuZ9/giphy.gif)
